File: PrivateCloudDisk-ios/PrivateCloudDisk-ios/Models/APIResponse.swift
//
//  APIResponse.swift
//  PrivateCloudDisk-ios
//
//  统一 API 响应模型
//

import Foundation

/// 统一 API 响应结构
struct APIResponse<T: Codable>: Codable {
    let code: Int
    let message: String?
    let data: T?
}

/// 空数据响应（仅需要 code + message 的接口）
struct APIEmptyResponse: Codable {
    let code: Int
    let message: String?
}

/// 分页数据
struct PageData<T: Codable>: Codable {
    let items: [T]
    let total: Int
    let page: Int
    let size: Int
}

/// WebSocket 消息类型
struct WSMessage<T: Codable>: Codable {
    let type: String
    let data: T?
    let timestamp: Int64?
    let senderId: String?
}

File: PrivateCloudDisk-workflow-service/src/main/java/org/project/workflow/model/ApiResponse.java
package org.project.workflow.model;

/** 新服务统一响应信封，稳定字符串 code 便于多客户端兼容。 */
public record ApiResponse<T>(String code, String message, T data, String requestId) {
    public static <T> ApiResponse<T> ok(T data, String requestId) {
        return new ApiResponse<>("OK", "操作成功", data, requestId == null ? "" : requestId);
    }
}
